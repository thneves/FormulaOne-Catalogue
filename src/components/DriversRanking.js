import PropTypes from 'prop-types';
import '../styles/components/DriversRanking.scss';
import defaultJohn from '../assets/images/defaultjohn.png';

const DriversRanking = ({
  driver,
  driverImg,
  position,
  points,
  wins,
  teamLogo,
}) => (
  <>
    <div className="driver-card">
      <span className="driver-position">
        { position }
      </span>
      <img className="driver-image" onError={e => { e.target.src = defaultJohn; }} src={driverImg} alt="driver profile" />
      <div className="driver-info">
        <span className="driver-name">
          {driver}
        </span>
        <span className="points-wins">
          Points&nbsp;&nbsp;
          <strong>{ points }</strong>
        </span>
        <span className="points-wins">
          Wins&nbsp;&nbsp;&nbsp;&nbsp;
          <strong>{ wins }</strong>
        </span>
      </div>
      <img className="team-logo" src={teamLogo} alt="team logo" />
    </div>
  </>
);

DriversRanking.defaultProps = {
  points: 0,
  wins: 0,
};

DriversRanking.propTypes = {
  driver: PropTypes.string.isRequired,
  driverImg: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired,
  points: PropTypes.number,
  teamLogo: PropTypes.string.isRequired,
  wins: PropTypes.number,
};

export default DriversRanking;
