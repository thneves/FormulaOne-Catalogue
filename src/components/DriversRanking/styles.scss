.driver-card {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 120px;
  width: 50%;
  padding: 2rem;
  margin: 2rem auto;
  padding: 0 1rem;
  border: 2px solid azure;
  border-radius: 10px;
  color: black;
  background-color: white;
  font-family: 'Orbitron', sans-serif;
}

.driver-position {
  font-size: 4rem;
  width: 3rem;
  text-align: center;
}

.driver-image {
  width: 80px;
  height: 88px;
  border-radius: 4px;
  border: 1px solid black;
}

.driver-info {
  display: flex;
  flex-direction: column;
  width: 30%;
  text-align: left;

  span {
    margin: 4px 0;
  }

  .driver-name {
    font-size: 1.2rem;
    letter-spacing: 1px;
    font-weight: 700;
  }

}

.team-logo {
  width: 175px;
  height: 88px;
}

