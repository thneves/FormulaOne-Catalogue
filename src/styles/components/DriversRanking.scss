.driver-card {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 120px;
  width: 50%;
  max-width: 620px;
  padding: 0 1rem;
  margin: 2rem auto;
  border: 2px solid azure;
  border-radius: 10px;
  color: black;
  background-color: white;
  font-family: "Orbitron", sans-serif;
  box-shadow:
    5px 5px 30px 7px rgba(0, 0, 0, 0.25),
    -5px -5px 30px 7px rgba(0, 0, 0, 0.22);
  transition: 0.7s;

  &:hover {
    color: white;
    background-color: transparent;
    border-radius: 8px;
    width: 47%;
    height: 115px;
    cursor: crosshair;
    box-shadow:
      5px 5px 30px 7px rgb(223, 213, 213),
      -5px -5px 30px 7px rgba(223, 217, 217, 0.925);
  }
}

.driver-position {
  font-size: 4rem;
  width: 3rem;
  text-align: center;
}

.driver-image {
  width: 80px;
  height: 88px;
  border-radius: 4px;
  border: 1px solid black;
  margin: 0 1.3rem;

  &:hover {
    cursor: pointer;
    border: 4px solid azure;
    border-radius: 15px;
  }
}

.driver-info {
  display: flex;
  flex-direction: column;
  width: 30%;
  text-align: left;

  span {
    margin: 4px 0;
  }

  .driver-name {
    font-size: 1.2rem;
    letter-spacing: 1px;
    font-weight: 700;
  }
}

@media (max-width: 860px) {
  .driver-card {
    width: 80%;
  }
}

@media (max-width: 596px) {
  .driver-card {
    width: 80%;
    height: 90px;
    padding: 0.8rem;
    margin: 1rem auto;
  }

  .driver-position {
    font-size: 2.2rem;
    width: 1rem;
    margin: 2px;
  }

  .driver-image {
    width: 70px;
    height: 78px;
    margin: 0 2px;
  }

  .driver-info {
    width: 38%;

    .driver-name {
      font-size: 0.8rem;
      letter-spacing: 0.4px;
    }

    .points-wins {
      font-size: 0.7rem;
    }
  }

  .team-logo {
    width: 95px;
    height: 68px;
  }
}

@media (max-width: 380px) {
  .driver-card {
    width: 70%;
  }

  .driver-image {
    display: none;
  }

  .driver-position {
    width: 1.5rem;
  }
}
