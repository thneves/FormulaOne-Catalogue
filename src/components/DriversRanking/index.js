import PropTypes from 'prop-types';
import './styles.scss';

const DriversRanking = ({
  driver,
  driverImg,
  position,
  points,
  wins,
  teamLogo,
}) => (
  <>
    <div className="driver-card">
      <span className="driver-position">
        { position }
      </span>
      <img className="driver-image" src={driverImg} alt="driver profile" />
      <div className="driver-info">
        <span className="driver-name">
          {driver}
        </span>
        <span>
          Points&nbsp;&nbsp;
          <strong>{ points }</strong>
        </span>
        <span>
          Wins&nbsp;&nbsp;&nbsp;&nbsp;
          <strong>{ wins }</strong>
        </span>
      </div>
      <img className="team-logo" src={teamLogo} alt="team logo" />
    </div>
  </>
);

DriversRanking.propTypes = {
  driver: PropTypes.string.isRequired,
  driverImg: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired,
  points: PropTypes.number.isRequired,
  teamLogo: PropTypes.string.isRequired,
  wins: PropTypes.number.isRequired,
};

export default DriversRanking;
