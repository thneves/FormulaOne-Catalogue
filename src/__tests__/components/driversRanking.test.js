import React from 'react';
import renderer from 'react-test-renderer';
import DriversRanking from '../../components/DriversRanking';

const driver = {
  name: 'John Doe',
  image: 'driver image',
  position: 1,
  points: 999,
  wins: 10,
  teamLogo: 'team logo image',

};

it('Renders Drivers Ranking', () => {
  const ranking = renderer
    .create(
      <DriversRanking
        position={driver.position}
        driver={driver.name}
        driverImg={driver.image}
        points={driver.points}
        wins={driver.wins}
        teamLogo={driver.teamLogo}
      />,
    )
    .toJSON();

  expect(ranking).toMatchSnapshot();
});
