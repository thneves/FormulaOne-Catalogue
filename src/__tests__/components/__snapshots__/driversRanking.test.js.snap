// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Drivers Ranking 1`] = `
<div
  className="driver-card"
>
  <span
    className="driver-position"
  >
    1
  </span>
  <img
    alt="driver profile"
    className="driver-image"
    onError={[Function]}
    src="driver image"
  />
  <div
    className="driver-info"
  >
    <span
      className="driver-name"
    >
      John Doe
    </span>
    <span
      className="points-wins"
    >
      Points  
      <strong>
        999
      </strong>
    </span>
    <span
      className="points-wins"
    >
      Wins    
      <strong>
        10
      </strong>
    </span>
  </div>
  <img
    alt="team logo"
    className="team-logo"
    src="team logo image"
  />
</div>
`;
